services/streammanager/kafka/kafkamanager.go

```diff
@@ -414,11 +456,27 @@ func prepareBatchOfMessages(topic string, batch []map[string]interface{}, timest
 			pkgLogger.Errorf("batch from topic %s is missing the userId attribute", topic)
 			continue
 		}
-
 		marshalledMsg, err := json.Marshal(message)
 		if err != nil {
 			return nil, err
 		}
+		codecs := p.getCodecs()
+		if len(codecs) > 0 {
+			schemaId, _ := data["schemaId"].(string)
+			fmt.Println(schemaId)
+			if schemaId == "" {
+				return nil, fmt.Errorf("schemaId is not available for this event")
+			}
+			if codec, ok := codecs[schemaId]; ok {
+				marshalledMsg, err = serialize(marshalledMsg, codec)
+				if err != nil {
+					return nil, fmt.Errorf("unable to convert the event with schemaId: %v, with error: %s", schemaId, err)
```

Comment:
Same here and possible in the rest of this function. Some info to pinpoint the event would be great. If available a `messageId` or otherwise the array position. WDYT?

Reply:
Updated the error message with messageId.